frame/society/src/lib.rs

```diff
@@ -1081,188 +1143,194 @@ pub mod pallet {
 		/// If a suspended member is rejected, remove all associated storage items, including
 		/// their payouts, and remove any vouched bids they currently have.
 		///
-		/// The dispatch origin for this call must be from the _SuspensionJudgementOrigin_.
+		/// The dispatch origin for this call must be Signed from the Founder.
 		///
 		/// Parameters:
 		/// - `who` - The suspended member to be judged.
 		/// - `forgive` - A boolean representing whether the suspension judgement origin forgives
 		///   (`true`) or rejects (`false`) a suspended member.
 		///
-		/// # <weight>
 		/// Key: B (len of bids), M (len of members)
-		/// - One storage read to check `who` is a suspended member. O(1)
-		/// - Up to one storage write O(M) with O(log M) binary search to add a member back to
-		///   society.
-		/// - Up to 3 storage removals O(1) to clean up a removed member.
-		/// - Up to one storage write O(B) with O(B) search to remove vouched bid from bids.
-		/// - Up to one additional event if unvouch takes place.
-		/// - One storage removal. O(1)
-		/// - One event for the judgement.
-		///
 		/// Total Complexity: O(M + logM + B)
-		/// # </weight>
 		#[pallet::weight(T::BlockWeights::get().max_block / 10)]
 		pub fn judge_suspended_member(
 			origin: OriginFor<T>,
 			who: T::AccountId,
 			forgive: bool,
-		) -> DispatchResult {
-			T::SuspensionJudgementOrigin::ensure_origin(origin)?;
-			ensure!(<SuspendedMembers<T, I>>::contains_key(&who), Error::<T, I>::NotSuspended);
+		) -> DispatchResultWithPostInfo {
+			let founder = ensure_signed(origin)?;
+			ensure!(Founder::<T, I>::get() == Some(founder.clone()), Error::<T, I>::NotFounder);
```

Comment:
Nit: `ensure!(Some(ensure_signed(origin)?) == Founder::<T, I>::get(), Error::<T, I>::NotFounder);`
Also in `bestow_membership()` and `kick_candidate()`

Reply:
Aside from the `.clone()` which may be unnecessary, I think this code is more readable - by convention we have a single line which checks the origin and extracts any value associated with it at the top. I wouldn't want to bury it in a large expression.